src/main/diagnostics/diagnostics.ts

```diff
@@ -1,68 +1,121 @@
-import DIAGNOSTICS_TABLE from '@/assets/diagnostics.yaml';
+export enum DiagnosticsCategory {
+  Utilities = 'Utilities',
+  Networking = 'Networking',
+}
+
+/**
+ * DiagnosticsChecker describes an implementation of a single diagnostics check.
+ */
+export interface DiagnosticsChecker {
+  /** Unique identifier for this check. */
+  id: string;
+  documentation: string,
+  description: string,
+  category: DiagnosticsCategory,
+  /**
+   * Preform the check.  Returns true if the check passed (i.e. things are
+   * working correctly and does not need to be fixed).
+   */
+  check(): Promise<boolean>;
+}
 
 type DiagnosticsFix = {
-  fixes: { description: string }
+  description: string
 };
 
-export type DiagnosticsCheck = {
+export type DiagnosticsResult = {
   id: string,
   documentation: string,
   description: string,
-  category: string,
+  category: DiagnosticsCategory,
+  passed: boolean,
   mute: boolean,
   fixes: DiagnosticsFix[],
 };
 
-type DiagnosticsCategories = {
-  title: string,
-  checks: Array<DiagnosticsCheck>,
-};
-
-type DiagnosticsType = {
-  last_update: string
-  categories: Array<DiagnosticsCategories>,
+export type DiagnosticsResultGroup = {
+  last_update: string,
+  checks: Array<DiagnosticsResult>,
 };
 
 export class Diagnostics {
-  diagnostics: DiagnosticsType;
-  checksByCategory: Record<string, Array<DiagnosticsCheck>> = {};
-  checks: Array<DiagnosticsCheck> = [];
-  constructor(diagnosticsTable: DiagnosticsType|undefined = undefined) {
-    this.diagnostics = diagnosticsTable || DIAGNOSTICS_TABLE.diagnostics;
-    for (const category of this.diagnostics.categories) {
-      for (const check of category.checks) {
-        check.mute ??= false;
-        check.fixes ??= [];
-        check.category = category.title;
-        this.checks.push(check);
+  /** Checkers capable of running individual diagnostics. */
+  checkers: Promise<DiagnosticsChecker[]>;
+
+  /** Time stamp of when the last check occurred. */
+  lastUpdate = new Date(0);
+
+  /** Last known check results. */
+  results: Array<DiagnosticsResult> = [];
+
+  /** Mapping of category name to diagnostic ids */
+  categories: Partial<Record<DiagnosticsCategory, string[]>> = {};
```

Comment:
Renamed it as `checkerIdByCategory`. This is only ever changed in the constructor, so I've tagged it `readonly` to avoid accidentally changing it. Fixed in da002f7c1f0fb21191d9e513689e674546cd5074.